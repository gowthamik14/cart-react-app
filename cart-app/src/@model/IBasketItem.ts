import IProduct from './IProduct';

export default interface IBasketItem extends IProduct {
  quantity: number;
}
