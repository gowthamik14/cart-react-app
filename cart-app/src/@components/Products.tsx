import React from 'react';
import OrderItem from './OrderItem';
import IProduct from '@model/IProduct';
import IOrderItem from '@model/IOrderItem';
import Product from './Product';
import IBasketItem from '@model/IBasketItem';

const Products = ({
  products,
  handleAddProduct,
  productsAddedToBasket
}: {
  products: IProduct[];
  handleAddProduct: (item: IProduct) => void;
  productsAddedToBasket: IBasketItem[]
}): JSX.Element => {
  return (
    <ul>
      {products.map((product) => (
        <Product
          key={product.id}
          id={product.id}
          name={product.name}
          price={product.price}       
          handleAddProduct={handleAddProduct} 
          allowAdd= {productsAddedToBasket.find(i=> i.id === product.id) ? false : true}        
        />
      ))}
    </ul>
  );
};

export default Products;
