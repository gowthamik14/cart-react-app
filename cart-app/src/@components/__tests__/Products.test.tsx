import React from 'react';
import { render} from '@testing-library/react';
import Products from '@components/Products';
import IBasketItem from '@model/IBasketItem';
import IProduct from '@model/IProduct';


test('renders learn react link', () => {
  const products: IProduct[] = [
    {
      name: 'cheese',
      price: 20,
      id: 1,
    },
    {
      name: 'Butter',
      price: 20,
      id: 1,
    },
  ];

  const productsAddedToBasket: IBasketItem[] = [
    {
      name: 'cheese',
      price: 20,
      id: 1,
      quantity: 0,
    },
    {
      name: 'Butter',
      price: 20,
      id: 1,
      quantity: 0,
    },
  ];

  const handleAddProduct = jest.fn();
  const { container } = render(
    <Products
      products={products}
      handleAddProduct={handleAddProduct}
      productsAddedToBasket= {productsAddedToBasket}
    />
  );

  expect(container).toMatchSnapshot();
  
});
