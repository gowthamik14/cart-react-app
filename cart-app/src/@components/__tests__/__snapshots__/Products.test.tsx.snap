// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <ul>
    <li
      class="flex justify-between items-center list-decimal list-inside border-b-2 border-gray-100	p-2"
    >
      <span>
        cheese
      </span>
      <div>
        <span
          class="text-gray-400 pr-1"
        >
          £
        </span>
        <span>
          0.20
        </span>
        <button
          class="ml-4 p-2 text-white rounded-md focus:outline-none focus:ring-2 ring-blue-200 ring-offset-2 bg-gray-400"
          disabled=""
        >
          Add
        </button>
      </div>
    </li>
    <li
      class="flex justify-between items-center list-decimal list-inside border-b-2 border-gray-100	p-2"
    >
      <span>
        Butter
      </span>
      <div>
        <span
          class="text-gray-400 pr-1"
        >
          £
        </span>
        <span>
          0.20
        </span>
        <button
          class="ml-4 p-2 text-white rounded-md focus:outline-none focus:ring-2 ring-blue-200 ring-offset-2 bg-gray-400"
          disabled=""
        >
          Add
        </button>
      </div>
    </li>
  </ul>
</div>
`;
